mineral-ui/mineral-ui
# [Select] [TextArea] [TextInput] Refactor to use FauxControl; fix z-index bug

### Description

- Refactor TextArea to use FauxControl, like Select & TextInput
- Extract height styles from FauxControl into SelectTrigger & TextInput, to keep the responsibility where it should be
- Fix z-index bug with TextArea when placed in a container with a background color
- Fix disabled text color bug in Safari (and many mobile Android browsers)
- Update snapshots of affected examples
- Always apply a background color (white, by default, but still configurable for cases like Card's examples) to the LivePreview, to catch z-index issues in the future

### Motivation and context

Discovered the z-index bug ([you can see the issue here](https://codesandbox.io/s/wk431361n7)) and decided to use the opportunity to refactor rather than merely applying 'zIndex: 1` to TextArea's root. Then discovered the disabled text color Safari bug when cross-browser testing, and fixed that, too.

### Screenshots, videos, or demo, if appropriate


https://textarea-z-index--mineral-ui.netlify.com/components/text-area

### How to test


1. `cd mineral-ui && npm start`
2. Navigate to Select, TextArea, & TextInput
3. Confirm all is well

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- Other (provide details below)

Refactor

### Checklist
